src/miniscript/types/extra_props.rs

```diff
@@ -119,6 +119,14 @@ pub struct ExtData {
     pub max_dissat_size: Option<(usize, usize)>,
     /// The timelock info about heightlocks and timelocks
     pub timelock_info: TimeLockInfo,
+    /// Maximum stack + alt stack size during satisfaction execution
+    /// This does **not** include initial witness elements. This element only captures
+    /// the additional elements that are pushed during execution.
+    pub exec_stack_elem_count_sat: Option<usize>,
+    /// Maximum stack + alt stack size during dissat execution
+    /// This does **not** include initial witness elements. This element only captures
+    /// the additional elements that are pushed during execution.
+    pub exec_stack_elem_count_dissat: Option<usize>,
```

Comment:
Why not include initial witness elements? I **think** every fragment has a fixed number of initial stack elements that it consumes before pushing new things onto the stack

Reply:
We are already tracking `max_elem_stack_count` for initial witness elements in a separate field. 

> I think every fragment has a fixed number of initial stack elements that it consumes before pushing new things onto the stack.

ors/ threshes can have multiple satisfactions which can different heights.

Reply:
Derp, yes, thank you!

Reply:
Wait, no, I'm still confused .. `or`s have multiple satisfactions but the max stack depth will just be the maximum of the children. Similar for `thresh`, you just take the sum of the `k` maximum-depth subitems.

Reply:
> but the max stack depth will just be the maximum of the children

If we include initial elements. The stack depth while executing `A` in and_b(A,B) would be `max_exec _A` + initial witness `A` + initial witness `B`. But for `B` would `max_exec_B` + `initial_witness_B` because A witness would have been consumed